packages/@o3r/core/schematics/add-modules/index.ts

```diff
@@ -0,0 +1,42 @@
+import { externalSchematic, Rule } from '@angular-devkit/schematics';
+import { NgAddModulesSchematicsSchema } from './schema';
+import { askQuestion } from '@angular/cli/src/utilities/prompt';
+import { getAvailableModulesWithLatestPackage, OTTER_MODULE_KEYWORD, OTTER_MODULE_SUPPORTED_SCOPES } from '@o3r/schematics';
+
+/**
+ * Select the available modules to add to the project
+ *
+ * @param _options
+ */
+export function ngAddModules(_options: NgAddModulesSchematicsSchema): Rule {
+  return async (tree, context) => {
+    if (!context.interactive) {
+      context.logger.error('This command is available only for interactive shell');
+      return;
```

Comment:
so that means we will never chain it with another rule? As far as I understood, you need to return the tree if it has been updated by another rule before